Add ability to filter urls by lastmod

Zak-Bahm opened the pull request:
This commit adds the config option `lastmod` that accepts a timestamp specify the minimum allowable `lastmod` value for urls.
Using this you can filter out older urls that may not be relevant anymore.
This should close this issue: https://github.com/seantomburke/sitemapper/issues/79

nagahshi commented:
+1 :100:

Zak-Bahm commented:
Any updates on this? It should be pretty straight-forward to merge but if there are some issues that need to be fixed I can handle them.

seantomburke commented:
@Zak-Bahm I think it looks good, just need to resolve the merge conflicts before it can be merged. I believe I tried, but didn't have permission to commit to the branch.

seantomburke commented on README.md:
```diff
@@ -62,6 +62,9 @@ sitemapper.fetch('https://wp.seantburke.com/sitemap.xml')
 
 You can add options on the initial Sitemapper object when instantiating it.
 
++ `requestHeaders`: (Object) - Additional Request Headers
++ `timeout`: (Number) - Maximum timeout for a single URL
```
@Zak-Bahm These are duplicated and not needed

seantomburke reviewed (approved):
Looks good, just needs merge conflicts resolved ![](https://i.shipit.today)

Zak-Bahm commented:
I removed the duplicate lines and fixed the conflicts so this should be good to go. I also bumped the version up to 3.2.4 but that can be changed if you think it should be a more major release like 3.3

seantomburke commented:
@Zak-Bahm Looks like the `package-lock.json` is broken and failing the tests, try removing `package-lock.json` then running `npm install -g npm` and then running `npm install` to fix the `package-lock.json`

Zak-Bahm commented:
Alright, that should be fixed.

Zak-Bahm commented:
@seantomburke could you review and approve the changes?

seantomburke commented:
Looks like it's still failing the tests

seantomburke commented:
@Zak-Bahm looks like the tests are still failing here, do they pass when you run it locally?

Zak-Bahm commented:
@seantomburke they were passing when I was running it locally but I wasn't testing older versions of npm like your automated tests do. I've tested back to npm v16.14.15 now so they should all run properly now.

Zak-Bahm commented:
@seantomburke is there anything on my end preventing this from being merged in?

seantomburke commented:
@Zak-Bahm There's a merge conflict with the `package-lock.json` file, It can't be merged because of this conflict. I tried fixing it on the `public-square:master` branch, so that would need to be merged into this branch in order to fix it. 
https://github.com/public-square/sitemapper/pull/2

seantomburke commented:
@Zak-Bahm This is what I see, I'm not able to merge it until the conflicts are resolved.
<img width="927" alt="Screen Shot 2022-07-28 at 11 52 26 PM" src="https://user-images.githubusercontent.com/965298/181700629-d0aeaa86-73ba-4080-8e43-a415c1795a7a.png">

seantomburke commented:
@Zak-Bahm Cloned this PR and fixed the merge conflict. Merged in https://github.com/seantomburke/sitemapper/pull/107, closing this one out!
